app/uk/gov/hmrc/apiplatform/modules/submissions/controllers/CredentialsRequestedController.scala

```diff
@@ -30,52 +30,23 @@ import javax.inject.{Inject, Singleton}
 import scala.concurrent.ExecutionContext
 import uk.gov.hmrc.thirdpartydeveloperfrontend.controllers.ApplicationController
 import uk.gov.hmrc.thirdpartydeveloperfrontend.controllers.checkpages.CanUseCheckActions
-import cats.data.NonEmptyList
-import uk.gov.hmrc.apiplatform.modules.submissions.domain.models._
 import uk.gov.hmrc.apiplatform.modules.submissions.services.SubmissionService
 import uk.gov.hmrc.apiplatform.modules.common.services.EitherTHelper
 import uk.gov.hmrc.thirdpartydeveloperfrontend.domain.models.controllers.BadRequestWithErrorMessage
 import uk.gov.hmrc.apiplatform.modules.submissions.controllers.models.AnswersViewModel._
+import uk.gov.hmrc.thirdpartydeveloperfrontend.domain.models.apidefinitions.APISubscriptionStatus
+import uk.gov.hmrc.apiplatform.modules.submissions.domain.models.Submission
+import uk.gov.hmrc.thirdpartydeveloperfrontend.domain.models.applications.Application
+import uk.gov.hmrc.apiplatform.modules.submissions.domain.models.AskWhen
 
 object CredentialsRequestedController {
-  case class ViewQuestion(id: QuestionId, text: String, answer: String)
-  case class ViewQuestionnaire(label: String, state: String, id: QuestionnaireId, questions: NonEmptyList[ViewQuestion])
-  case class ViewModel(appId: ApplicationId, appName: String, questionnaires: List[ViewQuestionnaire])
-
-  def convertAnswer(answer: ActualAnswer): Option[String] = answer match {
-    case SingleChoiceAnswer(value) => Some(value)
-    case TextAnswer(value) => Some(value)
-    case MultipleChoiceAnswer(values) => Some(values.mkString)
-    case NoAnswer => Some("n/a")
-    case AcknowledgedAnswer => None
-  }
-  
-  def convertQuestion(extSubmission: ExtendedSubmission)(item: QuestionItem): Option[ViewQuestion] = {
-    val id = item.question.id
-    
-    extSubmission.submission.latestInstance.answersToQuestions.get(id).flatMap(convertAnswer).map(answer =>
-      ViewQuestion(id, item.question.wording.value, answer)
-    )
-  }
-  
-  def convertQuestionnaire(extSubmission: ExtendedSubmission)(questionnaire: Questionnaire): Option[ViewQuestionnaire] = {
-    val progress = extSubmission.questionnaireProgress.get(questionnaire.id).get
-    val state = QuestionnaireState.describe(progress.state)
-    
-    val questions = questionnaire.questions
-      .map(convertQuestion(extSubmission))
-      .collect { case Some(x) => x }
-    NonEmptyList.fromList(questions)
-      .map(ViewQuestionnaire(questionnaire.label.value, state, questionnaire.id, _))
-  }
-
-  def convertSubmissionToViewModel(extSubmission: ExtendedSubmission)(appId: ApplicationId, appName: String): ViewModel = {
-    val questionnaires = extSubmission.submission.groups.flatMap(g => g.links)
-      .map(convertQuestionnaire(extSubmission))
-      .collect { case Some(x) => x }
-
-    ViewModel(appId, appName, questionnaires)
-  }
+  case class CredentialsRequestedViewModel (
```

Comment:
Would probably just call this ViewModel as the controller object provides enough context to know what type of view model it is. The same pattern we have used in the approvals repo.